unix.cpp

```diff
@@ -2,15 +2,59 @@
 #include "settings.h"
 #include <QDir>
 #include <QStandardPaths>
+#include <QString>
 
 namespace Sys {
 QString archiveName(void)
 {
     return "linux64.zip";
 }
 
+void migrateHomePath(void)
+{
+    QString legacyHomePath = QDir::homePath() + "/.unvanquished";
+    QString xdgDataHome = QStandardPaths::writableLocation(QStandardPaths::GenericDataLocation);
+    QString xdgHomePath = xdgDataHome + "/unvanquished";
+
+    if (QDir(legacyHomePath).exists()) {
```

Comment:
Personally, I think the code is more readable with fewer nested if statements, and just break the third condition onto its own line.

// Only migrate the homepath if the old legacy homepath exists, and the new ones don't yet exist.
if (QDir(legacyHomePath).exists() && !QDir(xdgHomePath).exists() 
    && !QDir().mkdir(xdgDataHome)) {
  qFatal("Could not create XDG data directory %s", qPrintable(xdgDataHome));
}

Reply:
I just modified the code to avoid some useless nesting (and removed a test that was just redoing something just tested).